### ISSUE TITLE
Start/end time logged incorrectly when submitting

### ISSUE THREAD

**Issue by asztal** [carstengehling/jirastopwatch#26]
I don't know if this is a limitation of JIRA's REST API, but I've noticed that when I submit a worklog, it will set the start time of the worklog to be the time that it's submitted. However, I think it would make more sense to set the end time of the worklog to be the time that it's submitted.

E.g. If I start working on a task at 9:30 and spend two hours on it, I submit it at 11:30 and the worklog gets created for 11:30 → 13:30. I think it would make more sense for to create it as 09:30 → 11:30.

**Comment by Nellemandela** [comment]
The api should support what you are suggesting

https://docs.atlassian.com/jira/REST/latest/#api/2/issue-addWorklog

{
    "comment": "I did some work here.",
    "started": "2016-07-11T010:00:00.000+0000",
    "timeSpentSeconds": 7200
}

**Comment by carstengehling** [comment]
Sounds like a good idea - but it's not really something that has bothered me. But feel free to submit a pull-request. :-)

**Comment by asztal** [comment]
Sounds good to me, I will have a look at it when I have some spare time :)

**Comment by carstengehling** [comment]
Just a note: StopWatch has no info about when you actually started the task - eg. if you've been switching between 2-3 tasks within the last couple of hours, it does not know, when you originally pressed "start" on the specific tasks.

So the best you will be able to do is StartTime = DateTime.Now - <time spent> - which is also the way I read your OP.

**Comment by asztal** [comment]
Yes, I figured that would be the case. In my case that's Good Enough™
because I generally submit a worklog when switching tasks. When I don't
it's usually because of a small distraction and the impact of those isn't
so great.

The main reason this will be useful to me is actually that sometimes I
forget to start tracking time against a task. Currently when I look in my
timesheet it's sometimes hard to see when I finished my last task because
the end times aren't what I expect, so having at least the correct end time
will help.
